chainstate/test-suite/src/tests/double_spend_tests.rs

```diff
@@ -432,6 +432,35 @@ fn duplicate_tx_in_the_same_block(#[case] seed: Seed) {
     });
 }
 
+#[rstest]
+#[trace]
+#[case(Seed::from_entropy())]
+fn duplicate_odd_tx_in_the_same_block(#[case] seed: Seed) {
+    utils::concurrency::model(move || {
+        let mut tf = TestFramework::default();
+
+        let mut rng = make_seedable_rng(seed);
+        let first_tx = tx_from_genesis(&tf.genesis(), &mut rng, 1);
+        let second_tx = tx_from_tx(&first_tx, rng.gen_range(1000..2000));
+        let third_tx = tx_from_tx(&second_tx, rng.gen_range(1000..2000));
+
+        let block = tf
+            .make_block_builder()
+            .with_transactions(vec![first_tx, second_tx, third_tx.clone(), third_tx])
```

Comment:
I would innovate a little more in such a test, picking a random tx out of the three, and inserting it in a random position.

Reply:
I wanted last 2 leaf of merkle tree to be identical to see how we handle this.

Reply:
Or... try all possibilities over time.

Reply:
I'll do it and push to save time this morning.